azkaban-common/src/main/java/azkaban/imagemgmt/daos/RampRuleDaoImpl.java

```diff
@@ -138,7 +147,29 @@ public Set<String> getOwners(final String ruleName) {
       return new HashSet<>(Arrays.asList(owners.split(",")));
     } catch (SQLException e) {
       LOG.error("Unable to fetch the hp flow ownership", e);
-      throw new ImageMgmtDaoException("failed to fetch hp flow owners into DB.");
+      throw new ImageMgmtDaoException("failed to fetch hp flow owners into DB." + e);
+    }
+  }
+
+  /**
+   * Get Unique ramp rule based on given ruleName.
+   * Fetched results would be converted into model {@link ImageRampRule}
+   *
+   * @param ruleName
+   * @return the ramp rule.
+   */
+  @Override
+  public ImageRampRule getRampRule(String ruleName) {
+    try {
+      List<ImageRampRule> rampRules = databaseOperator.query(FetchRampRuleHandler.FETCH_RAMP_RULE_BY_ID,
```

Comment:
Sorry for late question. In which case FETCH_RAMP_RULE_BY_ID can return more than 1 row?
If it is not possible, then I would recommend simplifying its handler to return "null" when no result or ImageRampRule instead of a collection.

Reply:
No ramp rule should be unique. I will change the handler